lib/segment/src/index/field_index/map_index.rs

```diff
@@ -175,6 +178,132 @@ impl<N: Hash + Eq + Clone + Display + FromStr> MapIndex<N> {
+        // Total points: 100
+        // Total values: 200
+        // total_excluded_value_count = 180
+        // non_excluded_values_count = 200 - 180 = 20
+        // max_values_per_point = 5 - 3 = 2
+        // min_not_excluded_by_values = 20 / 2 = 10
+        // min = max(10, 100 - 180) = 10
+        // exp = ...
+        // max = min(60, 20) = 20
+
+        let excluded_value_counts: Vec<_> = excluded
+            .iter()
+            .map(|val| self.map.get(val).map(|points| points.len()).unwrap_or(0))
+            .collect();
+        let total_excluded_value_count: usize = excluded_value_counts.iter().sum();
+
+        debug_assert!(total_excluded_value_count <= self.values_count);
+
+        let non_excluded_values_count =
+            self.values_count.saturating_sub(total_excluded_value_count);
+        let max_values_per_point = self.map.len().saturating_sub(excluded.len());
+
+        if max_values_per_point == 0 {
+            // All points are excluded, so we can't select any point
+            debug_assert_eq!(non_excluded_values_count, 0);
+            return CardinalityEstimation {
+                primary_clauses: vec![],
+                min: 0,
+                exp: 0,
+                max: 0,
+            };
+        }
+
+        // Minimal amount of points, required to fit all unused values.
+        // Cardinality can't be less than this value.
+        let min_not_excluded_by_values = div_ceil(non_excluded_values_count, max_values_per_point);
+
+        let min = min_not_excluded_by_values.max(
+            self.indexed_points
+                .saturating_sub(total_excluded_value_count),
+        );
+
+        // Maximum scenario: selected points overlap as much as possible.
+        // From one side, all excluded values should be assigned to the same point
+        // => we can take the value with the maximum amount of points.
+        // From another side, all other values should be enough to fill all other points.
+
+        let max_excluded_value_count = excluded_value_counts.iter().max().copied().unwrap_or(0);
+
+        let max = self
+            .indexed_points
+            .saturating_sub(max_excluded_value_count)
+            .min(non_excluded_values_count);
+
+        // Expected case: we assume that all points are filled equally.
+        // So we can estimate the probability of the point to have non-excluded value.
+        let exp = number_of_selected_points(self.indexed_points, non_excluded_values_count)
+            .max(min)
+            .min(max);
+
+        CardinalityEstimation {
+            primary_clauses: vec![],
+            min,
+            exp,
+            max,
+        }
+    }
+
+    fn except_iterator<'a>(
+        &'a self,
+        excluded: &'a [N],
+    ) -> Box<dyn Iterator<Item = PointOffsetType> + 'a> {
+        let iter = self
+            .map
+            .keys()
+            .filter(move |key| !excluded.contains(*key))
+            .flat_map(move |key| self.get_iterator(key))
```

Comment:
`move` is not required here:
```suggestion
            .filter(|key| !excluded.contains(*key))
            .flat_map(|key| self.get_iterator(key))
```

Reply:
Nice spotting, I thought Clippy would mention this.